packages/l/libsdl3/xmake.lua

```diff
@@ -0,0 +1,122 @@
+package("libsdl3")
+    set_homepage("https://www.libsdl.org/")
+    set_description("Simple DirectMedia Layer")
+    set_license("zlib")
+
+    if is_plat("mingw") and is_subhost("msys") then
+        add_extsources("pacman::SDL3")
+    elseif is_plat("linux") then
+        add_extsources("pacman::sdl3", "apt::libsdl3-dev")
+    elseif is_plat("macosx") then
+        add_extsources("brew::sdl3")
+    end
+
+    add_urls("https://www.libsdl.org/release/SDL3-$(version).zip",
+             "https://github.com/libsdl-org/SDL/releases/download/release-$(version)/SDL3-$(version).zip", { alias = "archive" })
+    add_urls("https://github.com/libsdl-org/SDL.git", { alias = "github" })
+
+    add_versions("archive:3.2.0", "abe7114fa42edcc8097856787fa5d37f256d97e365b71368b60764fe7c10e4f8")
+    add_versions("github:3.2.0", "release-3.2.0")
+
+    add_deps("cmake", "egl-headers", "opengl-headers")
+
+    if is_plat("linux", "bsd") then
+        add_configs("x11", {description = "Enables X11 support (requires it on the system)", default = true, type = "boolean"})
+        add_configs("wayland", {description = "Enables Wayland support", default = true, type = "boolean"})
```

Comment:
I don't do linux development, but is there a reason why these are enabled by default on that platform? I thought you could develop a libsdl3 application without these?

Reply:
One of the main feature of SDL (and reason people use it) is input and window management, which need x11/wayland to work.
It's very rare to use libsdl without these

Reply:
I'm always wary about defaults being enabled because you can't really take it back later. Even if it's rare, minimal dependencies by default should still be a consideration.

Reply:
I understand, but libsdl is a bit special, it only requires headers to build as it doesn't link x11 nor wayland, it loads them manually (dlopen/dlsym) at runtime only if they're available.

A good default for libsdl is to build it with as many deps as possible for this reason